geode-core/src/main/java/org/apache/geode/metrics/internal/InternalDistributedSystemMetricsService.java

```diff
@@ -227,6 +227,7 @@ private void clearAndCloseMeterRegistry() {
       new HashSet<>(meterRegistry.getRegistries())
           .forEach(meterRegistry::remove);
 
+      meterRegistry.clear();
```

Comment:
If you call clear first, then when close is called it will not be able to ask each thing in the registry to close. If calling close on each meter is important then you should call clear AFTER you call close,

Reply:
At the moment, every meter we add uses the default `close()`, which does nothing.

That said, I don't want to _count_ on meters doing nothing during `close()`. So I'll change it.